src/System.Numerics.Vectors/src/System/Numerics/VectorMath.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+#if HAS_INTRINSICS
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace System.Numerics
+{
+    internal static class VectorMath
+    {
+        public static Vector128<float> Lerp(Vector128<float> a, Vector128<float> b, Vector128<float> t)
+        {
+            if (Sse.IsSupported)
+            {
+                return Sse.Add(a, Sse.Multiply(Sse.Subtract(b, a), t));
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public static bool Equal(Vector128<float> vector1, Vector128<float> vector2)
+        {
+            if (Sse.IsSupported)
+            {
+                // do not use !Equal if vectors contain NaN
```

Comment:
You probably should use `CompareEqual` and `0xF` here (for `==`) since it properly handles `NaN` (reports `false` for `NaN`)

You can use `CompareNotEqual` for `!=`, where it properly handles `NaN` (reports `true` for `NaN`)

Reply:
@EgorBo, could you provide an example of where `Sse.CompareNotEqual(vector1, vector2)) != 0` fails for `!=`?

Reply:
![image](https://user-images.githubusercontent.com/523221/44238988-bb6b5880-a1bf-11e8-9491-209786a2e7e7.png)

Reply:
How is that incorrect? `v1 != v2` so `notEqual is true`

Reply:
@tannergooding sorry, wrong example. I meant {NaN,0,0,0} != {NaN,0,0,0}, the code above will return `true` but there is a test: https://github.com/dotnet/corefx/blob/master/src/System.Numerics.Vectors/tests/Matrix4x4Tests.cs#L2438-L2439

Reply:
@tannergooding now I get it. for some reason I was convinced that when NaN is involved anywhere - both == and != should always be false.

Reply:
Ah. Yeah, for `NaN`:
* == is always false
* != is always true